L/LAMMPS/build_tarballs.jl

```diff
@@ -58,12 +59,38 @@ products = [
     ExecutableProduct("lmp", :lmp),
 ]
 
+mpi_abis = (
+    (:mpich, PackageSpec(name="MPICH_jll"), "") ,
+    (:msmpi, PackageSpec(name="MicrosoftMPI_jll"), ""),
+    (:mpitrampoline, PackageSpec(name="MPItrampoline_jll"), "2")
+)
+
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll")),
-    Dependency(PackageSpec(name="MPItrampoline_jll"), compat="2"),
-    Dependency(PackageSpec(name="MicrosoftMPI_jll"))
+    # Dependency(PackageSpec(name="MPIPlatformTag")),
 ]
 
+all_platforms = AbstractPlatform[]
+for (abi, pkg, compat) in mpi_abis
+    pkg_platforms = deepcopy(platforms)
+    foreach(pkg_platforms) do p
+        BinaryPlatforms.add_tag!(p.tags, "mpi", string(abi))
+    end
+    append!(all_platforms, pkg_platforms)
+    push!(dependencies, Dependency(pkg; compat, platforms=pkg_platforms))
+end
+
+augmented_platform_block = """
+    using Base.BinaryPlatforms
+    # using MPIPlatformTag
+
+    function augmented_platform(platform)
```

Comment:
@staticfloat does this look okay in principle?

Reply:
Yeah, seems reasonable to me!  Let's try it.